app/src/main/java/org/fossasia/openevent/general/MainActivity.kt

```diff
@@ -84,6 +84,9 @@ class MainActivity : AppCompatActivity() {
                     mainFragmentCoordinatorLayout, R.string.sign_in_canceled, Snackbar.LENGTH_SHORT
                 ).show()
             }
+            R.id.searchFragment -> {
+                navController.popBackStack(R.id.eventsFragment,false)
+            }
```

Comment:
Wrong. Just use nav options to clear back stack when you are redirecting to search. No need to add for back pressed in Main Activity. Later it will contain bugs.

Reply:
@liveHarshit is their any existing method to clear back stack in navigation class ?

Reply:
Use the following while navigating to search fragment. And check if it works fine.

```java
val builder = NavOptions.Builder()
builder.setPopUpTo(R.id.eventsFragment, false)
builder.setEnterAnim(R.anim.slide_in_right)
builder.setExitAnim(R.anim.slide_out_left)
builder.setPopEnterAnim(R.anim.slide_in_left)
builder.setPopExitAnim(R.anim.slide_out_right)
Navigation.findNavController(rootView).navigate(R.id.searchFragment, null, builder.build())
```

Reply:
It worked.